ql/CMakeLists.txt

```diff
@@ -1,19 +1,28 @@
+
+get_quantlib_library_name(QL_OUTPUT_NAME)
+
+
 file(GLOB_RECURSE QUANTLIB_FILES "*.hpp" "*.cpp")
-add_library(QuantLib SHARED ${QUANTLIB_FILES})
-add_library(QuantLib_Static STATIC ${QUANTLIB_FILES})
+if(WIN32)
+    # Windows needs to link static library (nothing is declared to export in QuantLib)
+    add_library(${QL_OUTPUT_NAME} STATIC ${QUANTLIB_FILES})
+    set_property(TARGET ${QL_OUTPUT_NAME} PROPERTY PROJECT_LABEL "QuantLib")
+else()
+    add_library(${QL_OUTPUT_NAME} SHARED ${QUANTLIB_FILES})
```

Comment:
Windows need to compile it in STATIC mode (nothing is exported from the DLL).

UNIX is compiling it `SHARED` always... but I think it would be better to not specify it and let the user select it with the `BUILD_SHARED_LIBS` var, don't you think so?

Reply:
By default, Unix builds both shared and static, and the previous version did that.  But if the pattern in CMake is to choose one or the other through `BUILD_SHARED_LIBS`, I'm not necessarily against that.  Is the support there already, or do you have to add it?

Reply:
The flag `BUILD_SHARED_LIBS` is provided by CMake and [works out of the box](https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html) if you do not force static/shared explicitly. I would suggest:
 * Declare it explicit as an option.
 * Use it for Unix: user can choose to build shared or static. I think that people wants one or the other, but not both at the same time.
 * On Windows write a warning message if set to `True` because it will compile, but as nothing is exported, it has no sense to build it shared.